File: tests/fixtures/java/src/main/java/com/example/myapp/repository/UserRepository.java
package com.example.myapp.repository;

import java.util.List;
import java.util.ArrayList;
import com.example.myapp.model.User;

/**
 * User repository for data access.
 */
public class UserRepository {
    private List<User> users = new ArrayList<>();
    
    public List<User> findAll() {
        return new ArrayList<>(users);
    }
    
    public User findById(Long id) {
        for (User user : users) {
            if (user.getId().equals(id)) {
                return user;
            }
        }
        return null;
    }
    
    public void save(User user) {
        users.add(user);
    }
    
    public void delete(Long id) {
        users.removeIf(user -> user.getId().equals(id));
    }
}

File: tests/fixtures/kotlin/src/main/kotlin/com/example/myapp/repository/UserRepository.kt
package com.example.myapp.repository

import com.example.myapp.model.User

/**
 * User repository for data access.
 */
class UserRepository {
    private val users = mutableListOf<User>()
    
    fun findAll(): List<User> {
        return users.toList()
    }
    
    fun findById(id: Long): User? {
        return users.find { it.id == id }
    }
    
    fun save(user: User) {
        users.add(user)
    }
    
    fun delete(id: Long) {
        users.removeIf { it.id == id }
    }
}
